pysat/tests/classes/cls_instrument_property.py

```diff
@@ -0,0 +1,648 @@
+    def test_custom_instrument_load_incomplete(self, del_routine):
+        """Test if exception is thrown if supplied routines are incomplete.
+
+        Parameters
+        ----------
+        del_routine : str
+            Name of required routine to delete from module.
+
+        """
+
+        import pysat.instruments.pysat_testing as test
+        delattr(test, del_routine)
+
+        with pytest.raises(AttributeError) as aerr:
+            pysat.Instrument(inst_module=test, tag='',
+                             clean_level='clean')
+        estr = 'A `{:}` function is required'.format(del_routine)
+        assert str(aerr).find(estr) >= 0
+        return
+
+    @pytest.mark.parametrize("func, kwarg, val", [('init', 'test_init_kwarg',
+                                                   True),
+                                                  ('clean', 'test_clean_kwarg',
+                                                   False),
+                                                  ('preprocess',
+                                                   'test_preprocess_kwarg',
+                                                   'test_phrase'),
+                                                  ('load', 'test_load_kwarg',
+                                                   'bright_light'),
+                                                  ('list_files',
+                                                   'test_list_files_kwarg',
+                                                   'sleep_tight'),
+                                                  ('list_remote_files',
+                                                   'test_list_remote_kwarg',
+                                                   'one_eye_open'),
+                                                  ('download',
+                                                   'test_download_kwarg',
+                                                   'exit_night')
+                                                  ])
+    def test_instrument_function_keywords(self, caplog, func, kwarg, val):
+        """Test if Instrument function keywords are registered by pysat.
+
+        Parameters
+        ----------
+        func : str
+            Function name to test.
+        kwarg : str
+            Keyword argument of function to modify.
+        val : str
+            Replacement value for modified kwarg.
+
+        """
+
+        with caplog.at_level(logging.INFO, logger='pysat'):
+            # Trigger load functions
+            self.testInst.load(date=self.ref_time)
+
+            # Refresh files to trigger other functions
+            self.testInst.files.refresh()
+
+            # Get remote file list
+            self.testInst.download_updated_files()
+
+        # Confirm kwargs made it where they should be
+        assert kwarg in self.testInst.kwargs[func]
+        assert self.testInst.kwargs[func][kwarg] == val
+
+        # Check if function under test can assign attributes, not all can
```

Comment:
Could be out of scope here, but I think this comment is stale. I think @aburrell extended 'live' arguments to all functions after this was written.

Reply:
This `live_check` is repeated in a few of these tests.

Reply:
Going to leave as is.  Might be worth writing up an issue if this section of the tests is stale.

Reply:
Probably should write up an issue to look into this, at a minimum.

Reply:
Added #914